### PR TITLE
Add gh-pages rendered by gohugo

### PR DESCRIPTION
**Pull request by rgeyer**
This will add a github action which will trigger when a push is made to the master branch, containing changes to the `docs/**` path.

It publishes to a new branch named `gh-pages`. The first time the action runs, the branch is created, then the gh-pages setting must be changed, and a new deploy created as documented [here](https://github.com/peaceiris/actions-gh-pages#%EF%B8%8F-first-deployment-with-github_token).

This is a bit awkward, but I intentionally left a field in the `config.yaml` file unchanged so that it can be changed and committed after the gh-pages branch is created properly.

### CODE REVIEW

File: docs/config.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
+baseURL: https://rgeyer.github.io/hugoactionstest/
```
Comment (rgeyer) [inline docs/config.yaml:1]: After this PR is merged. Follow [these directions](https://github.com/peaceiris/actions-gh-pages#%EF%B8%8F-first-deployment-with-github_token) to point gh-pages to the newly created branch.

Then this line can be changed to `https://grafana.github.io/grizzly/` and pushed, which will trigger the final github action execution and *actual* deployment of the docs.

File: .github/workflows/gh-pages.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  push:
+    branches:
+      - master  # Set a branch to deploy
+    paths:
+      - 'docs/**'
+  pull_request:
+
+jobs:
+  deploy:
+    runs-on: ubuntu-18.04
```
Comment (malcolmholmes) [inline .github/workflows/gh-pages.yml:13]: can we use 20.04? Will last us longer?

File: docs/static/images/logo.svg [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+             id="path152" /></g></g><g
+         id="g154"><g
+           id="g156"
+           clip-path="url(#clipPath160)"><path
+             d="m 5965.71,4557.25 c 0,-158.68 -53.72,-305.28 -144.14,-421.82 0,-0.23 0,-0.23 0,-0.23 -126.15,-163.35 -323.99,-268.33 -546.23,-268.33 -98.32,0 -191.44,20.45 -276.2,57.42 243.92,106.68 414.41,349.87 414.41,632.96 0,4.93 0,9.61 -0.25,14.54 -5.65,276.93 -174.43,513.71 -414.16,618.42 -239.73,-104.71 -408.27,-341.49 -414.17,-618.42 -0.26,-4.69 -0.26,-9.61 -0.26,-14.54 0,-141.67 42.63,-273.24 115.81,-382.63 72.93,-109.4 176.65,-197.11 298.62,-250.33 -84.52,-36.97 -177.89,-57.42 -276.2,-57.42 -222.49,0 -420.09,104.98 -546.24,268.56 -90.17,116.54 -143.89,263.14 -143.89,421.82 0,10.59 0,20.7 0.5,31.05 1.71,21.91 3.69,43.6 6.15,65.03 69.73,649.73 435.62,1210.51 959.68,1544.84 524.3,-334.33 889.93,-895.11 959.92,-1544.84 0,0 0,-0.49 0,-0.73 2.21,-21.18 4.18,-42.62 5.9,-64.3 0.49,-10.35 0.75,-20.46 0.75,-31.05"
+             style="fill:url(#linearGradient166);fill-opacity:1;fill-rule:nonzero;stroke:none"
+             id="path168" /></g></g></g></g></svg>
```
Comment (malcolmholmes) [inline docs/static/images/logo.svg:271]: where did you get this logo from? I think it is great! We could also ask the design team for one.

**Review by malcolmholmes — APPROVED** [review]

File: docs/static/images/logo.svg (hunk 3 above)
Comment (rgeyer) [reply docs/static/images/logo.svg:271]: Free, and royalty-free stuff, just the first thing that matched my search and caught my eye.

https://www.rawpixel.com/image/524731/free-illustration-vector-bear-animal-art

File: .github/workflows/gh-pages.yml (hunk 2 above)
Comment (rgeyer) [reply .github/workflows/gh-pages.yml:13]: Sure! I'm a bit new to github actions, so this was largely copy/pasted, and this was just a byproduct of that.

**Pull request merged**